Unix/base/batch.h

```diff
@@ -113,68 +113,173 @@ void Batch_InitFromBuffer(
 void Batch_Destroy(
     Batch* self);
 
-Batch* Batch_New(
-    size_t maxPages);
-
 void Batch_Delete(
     Batch* self);
 
-void* Batch_Get(
+#if defined(CONFIG_ENABLE_DEBUG) && defined(CONFIG_OS_LINUX)
+
+#define BATCH_CALLSITE __FILE__, __LINE__, __FUNCTION__
+#define Batch_New(pages) __Batch_New(BATCH_CALLSITE, pages)
+#define Batch_Get(self, size) __Batch_Get(BATCH_CALLSITE, self, size)
+#define Batch_GetClear(self, size) __Batch_GetClear(BATCH_CALLSITE, self, size)
+#define Batch_Tcsdup(self, str) __Batch_Tcsdup(BATCH_CALLSITE, self, str)
+#define Batch_Strdup(self, str) __Batch_Strdup(BATCH_CALLSITE, self, str)
+#define Batch_StrTcsdup(self, str) __Batch_StrTcsdup(BATCH_CALLSITE, self, str)
+#define Batch_ZStrdup(self, str) __Batch_ZStrdup(BATCH_CALLSITE, self, str)
+
+Batch* __Batch_New(
+    const char* file,
+    size_t line,
+    const char* func,
+    size_t maxPages);
+
+void* __Batch_Get(
+    const char* file,
+    size_t line,
+    const char* func,
     Batch* self,
     size_t size);
 
-MI_INLINE void* Batch_GetClear(
+MI_INLINE void* __Batch_GetClear(
+    const char* file,
+    size_t line,
+    const char* func,
     Batch* self,
     size_t size)
 {
-    void* ptr = Batch_Get(self, size);
+    void* ptr = __Batch_Get(file, line, func, self, size);
 
     if (ptr)
         memset(ptr, 0, size);
 
     return ptr;
 }
 
-void Batch_Put(
+_Check_return_
+ZChar* __Batch_Tcsdup(
+    const char* file,
+    size_t line,
+    const char* func,
     Batch* self,
-    void* ptr);
+    const ZChar* str);
 
 _Check_return_
-ZChar* Batch_Tcsdup(
+char* __Batch_Strdup(
+    const char* file,
+    size_t line,
+    const char* func,
+    Batch* self,
+    const char* str);
+
+#if defined(CONFIG_ENABLE_WCHAR)
+ZChar* __Batch_StrTcsdup(
+    const char* file,
+    size_t line,
+    const char* func,
+    Batch* self,
+    const char* str);
+#else
+MI_INLINE ZChar* __Batch_StrTcsdup(
+    const char* file,
+    size_t line,
+    const char* func,
+    Batch* self,
+    const char* str)
+{
+    return __Batch_Strdup(file, line, func, self, str);
+}
+#endif
+
+#if defined(CONFIG_ENABLE_WCHAR)
```

Comment:
Is WCHAR still a supported build?

Reply:
We still have a lot of code based on CONFIG_ENABLE_WCHAR.  If you don't think we'll support it, I can remove it.